clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/store/preference/InAppStore.kt

```diff
@@ -167,25 +169,41 @@ class InAppStore(
     /**
      * Reads evaluated Server-side In-App IDs.
      *
-     * @return An array of evaluated Server-side In-App IDs.
+     * @return A JSoNObject representing the map of EventType - evaluated Server-side In-App IDs
      */
-    fun readEvaluatedServerSideInAppIds(): JSONArray {
+    fun readEvaluatedServerSideInAppIds(): JSONObject {
         val evaluatedServerSideInAppIds = ctPreference.readString(PREFS_EVALUATED_INAPP_KEY_SS, "")
-        if (evaluatedServerSideInAppIds.isNullOrBlank()) return JSONArray()
-
-        return JSONArray(evaluatedServerSideInAppIds)
+        if (evaluatedServerSideInAppIds.isNullOrBlank()) return JSONObject()
+
+        return try {
+            // Try to convert the string to a JSONObject
+            JSONObject(evaluatedServerSideInAppIds)
+        } catch (jsonException: JSONException) {
+            // If it fails, convert the string to a JSONArray
+            val jsonArray = JSONArray(evaluatedServerSideInAppIds)
+            // Wrap the JSONArray in a JSONObject
+            JSONObject().put(Constants.RAISED, jsonArray)
```

Comment:
so basically we expect the catch block to trigger first time and it correcting the data? 
is there no code in migration where are check version upgrade.

Reply:
Yess. That was the idea
In such a case, version check won't be needed

Reply:
@CTLalit addressed in latest commit